### PR TITLE
feat: Add category column to blob and transaction tables

### LINKED ISSUE #595: Add category column to transactions and blobs view tables

### PR DESCRIPTION
**Pull request by xFJA**
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [X] I have filled out the description and linked the related issues.

### Description
A `category` column has been added to `blob`, `block` and `transactions` tables at the init of the rows without any header name

### Related Issue
Closes #595 

## Transaction table
image

## Block table

https://github.com/user-attachments/assets/a932f886-d5b0-44a1-8c2f-68598a8513ef

### CODE REVIEW

**Review by PJColombo — COMMENTED** [review]
"Blob Gas Price" column on tx table looks cropped

image


Also, category is not being displayed on the blobs table:

image

**Comment by PJColombo** [comment]
I'm also thinking it would make more sense to remove "Rollup" column and display rollup logo in the "Category" by adding it right next to the "Rollup" label.

**Comment by xFJA** [comment]
> "Blob Gas Price" column on tx table looks cropped
> 
> image
> Also, category is not being displayed on the blobs table:
> 
> image



> "Blob Gas Price" column on tx table looks cropped
> 
> image
> Also, category is not being displayed on the blobs table:
> 
> image

- Fixed the category not displayed on `blobs` table 
- I've checked the `Blob Gas Price` column in `blobs` table and I cannot notice any crop. The table has a horizontal scroll for the overflowing columns. Could you check it again?
  image

**Comment by xFJA** [comment]
> I'm also thinking it would make more sense to remove "Rollup" column and display rollup logo in the "Category" by adding it right next to the "Rollup" label.

Agree! Having both columns is redundant and removes width space. I have merged both columns into one with your proposal 👍

**Review by PabloCastellano — APPROVED** [review]
lgtm :+1:

**Comment by PJColombo** [comment]
I'll comment here what we discussed on Discord:

Having both the category and icon feels too verbose. As an alternative, I propose the following:

We add a new column as the first column in all tables, leaving the title blank:
	•	If the category is ROLLUP, we display the icon; otherwise, the column remains empty.
	•	For the block table specifically, we display the icons for all transactions of the ROLLUP category, overlapping as needed.

**Comment by xFJA** [comment]
> I'll comment here what we discussed on Discord:
> 
> Having both the category and icon feels too verbose. As an alternative, I propose the following:
> 
> We add a new column as the first column in all tables, leaving the title blank: • If the category is ROLLUP, we display the icon; otherwise, the column remains empty. • For the block table specifically, we display the icons for all transactions of the ROLLUP category, overlapping as nee

> I'll comment here what we discussed on Discord:
> 
> Having both the category and icon feels too verbose. As an alternative, I propose the following:
> 
> We add a new column as the first column in all tables, leaving the title blank: • If the category is ROLLUP, we display the icon; otherwise, the column remains empty. • For the block table specifically, we display the icons for all transactions of the ROLLUP category, overlapping as needed.

Agree! Done 👍 

In `blocks` table:
- I added the rollup icons overlapped a little bit
- I added the rollup icon in the inner expand table for each specific transaction

File: apps/web/src/pages/txs.tsx [hunk 1]
```diff
@@ -202,6 +202,14 @@ const Txs: NextPage = function () {
 
           return {
             cells: [
+              {
+                item:
+                  category && category === "rollup" && rollup ? (
```
Comment (PJColombo) [inline apps/web/src/pages/txs.tsx:207]:
```suggestion
                 category === "rollup" && rollup ? (
```

File: apps/web/src/pages/blocks.tsx [hunk 2]
```diff
@@ -174,6 +190,24 @@ const Blocks: NextPage = function () {
...
           return {
             cells: [
+              {
+                item: (
+                  <div className="relative flex">
+                    {transactions.map((tx, i) => {
+                      return tx.rollup ? (
+                        <div
+                          key={i}
+                          className="-ml-1 -translate-y-1 scale-100 first-of-type:ml-0 hover:scale-125 "
```
Comment (PJColombo) [inline apps/web/src/pages/blocks.tsx:200]: I'd prefer to remove the scale animation to keep consistency with the rest of tables

File: apps/web/src/pages/blocks.tsx [hunk 3]
```diff
@@ -127,12 +133,22 @@ const Blocks: NextPage = function () {
...
                 }))
             );
 
             const rows = transactionsCombinedWithInnerBlobs.map(
-              ({ transactionHash, blobVersionedHash }) => ({
+              ({ transactionHash, blobVersionedHash, rollup, category }) => ({
                 cells: [
+                  {
+                    item:
+                      category && category === "rollup" && rollup ? (
```
Comment (PJColombo) [inline apps/web/src/pages/blocks.tsx:146]:
```suggestion
                      category === "rollup" && rollup ? (
```

File: apps/web/src/pages/blobs.tsx [hunk 4]
```diff
@@ -84,8 +89,19 @@ const Blobs: NextPage = function () {
...
             blockTimestamp,
             blockNumber,
+            transaction,
           }) => ({
             cells: [
+              {
+                item:
+                  transaction?.category &&
+                  transaction.category === "rollup" &&
+                  transaction.rollup ? (
```
Comment (PJColombo) [inline apps/web/src/pages/blobs.tsx:99]:
```suggestion
                  transaction?.category === "rollup" &&
                  transaction.rollup ? (
```

**Review by PJColombo — CHANGES_REQUESTED** [review]
Nice job.

I’ve left a few comments along with some small suggestions.

File: apps/web/src/pages/blocks.tsx (hunk 2 above)
Comment (xFJA) [reply apps/web/src/pages/blocks.tsx]: Done

File: apps/web/src/pages/blobs.tsx (hunk 4 above)
Comment (xFJA) [reply apps/web/src/pages/blobs.tsx]: Done

**Review by PJColombo — APPROVED** [review]
lgtm

**Pull request merged**